wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/EntityIdValueImpl.java

```diff
@@ -143,33 +144,47 @@ public static EntityIdValue fromId(String id, String siteIri) {
 	 *
 	 * @param id
 	 * 		the identifier of the entity, such as "Q42"
+	 * @param returnJsonEntity
+	 * 		returns JSON entity types when set to true
 	 * @throws IllegalArgumentException
 	 *      if the id is invalid
 	 */
-	static String guessEntityTypeFromId(String id) {
+	static String guessEntityTypeFromId(String id, boolean returnJsonEntity) {
 		if(id.isEmpty()) {
 			throw new IllegalArgumentException("Entity ids should not be empty.");
 		}
 		switch (id.charAt(0)) {
 			case 'L':
 				if(id.contains("-F")) {
-					return JSON_ENTITY_TYPE_FORM;
+					return returnJsonEntity ? JSON_ENTITY_TYPE_FORM : DatatypeIdValue.DT_FORM;
 				} else if(id.contains("-S")) {
-					return JSON_ENTITY_TYPE_SENSE;
+					return returnJsonEntity ? JSON_ENTITY_TYPE_SENSE : DatatypeIdValue.DT_SENSE;
 				} else {
-					return JSON_ENTITY_TYPE_LEXEME;
+					return returnJsonEntity ? JSON_ENTITY_TYPE_LEXEME : DatatypeIdValue.DT_LEXEME;
 				}
 			case 'M':
 				return JSON_ENTITY_TYPE_MEDIA_INFO;
```

Comment:
You probably want to return here `DatatypeIdValue.DT_MEDIA_INFO`, no?

Reply:
Yes, thank you! I've fixed it and pushed my changes.